Google.Api.Gax.Grpc.Tests/ClientBuilderBaseTest.GetGoogleCredentialTest.cs

```diff
@@ -0,0 +1,149 @@
+                {
+                    JsonCredentials = s_serviceAccountJson
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+            }
+
+            [Fact]
+            public async Task CredentialsPath_Valid()
+            {
+                var path = Path.GetTempFileName();
+                File.WriteAllText(path, s_serviceAccountJson);
+
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    CredentialsPath = path
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+                File.Delete(path);
+            }
+
+            [Fact]
+            public async Task GoogleCredential_Valid()
+            {
+                var credential = GoogleCredential.FromJson(s_serviceAccountJson);
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    GoogleCredential = credential
+                };
+                await AssertCredentialAsync(builder, AssertBasicProperties);
+            }
+
+            [Fact]
+            public async Task UseJwtAccessWithScopesFalse()
+            {
+                var credential = GoogleCredential.FromJson(s_serviceAccountJson);
+                var builder = new FakeBuilder(s_scopes, false)
+                {
+                    GoogleCredential = credential
+                };
+                await AssertCredentialAsync(builder,
+                    credential => Assert.False(((ServiceAccountCredential)credential.UnderlyingCredential).UseJwtAccessWithScopes));
+            }
+
+            [Fact]
+            public async Task QuotaProject()
+            {
+                var credential = GoogleCredential.FromJson(s_serviceAccountJson);
+                var builder = new FakeBuilder(s_scopes, true)
+                {
+                    GoogleCredential = credential,
+                    QuotaProject = "quota-project"
+                };
+                await AssertCredentialAsync(builder,
+                    credential => Assert.Equal("quota-project", ((ServiceAccountCredential) credential.UnderlyingCredential).QuotaProject));
+            }
+
+            private static void AssertBasicProperties(GoogleCredential credential)
+            {
+                var sa = Assert.IsType<ServiceAccountCredential>(credential.UnderlyingCredential);
+                Assert.Equal("CLIENT_EMAIL", sa.Id);
+                Assert.Equal(s_scopes, sa.Scopes);
+                Assert.True(sa.UseJwtAccessWithScopes);
+            }
+
+            private async Task AssertCredentialAsync(FakeBuilder builder, Action<GoogleCredential> assertion)
+            {
+                var cred1 = builder.GetGoogleCredential();
+                assertion(cred1);
+                var cred2 = await builder.GetGoogleCredentialAsync(default);
+                assertion(cred2);
+            }
```

Comment:
Maybe write a couple of theories for checking validation, one that succeeds and one that fails?